### ISSUE TITLE
WebUI not opening, error with long traceback of pydantic and starlette validation

### ISSUE THREAD

**Issue by Tremeschin** [BrokenSource/DepthFlow#39]
**In case people from FastAPI reads here**, v0.114.2 doesn't fix it for older versions of Gradio, my releases are broken

- **TLDR**; It's an [**regression with upstream fastapi**](https://github.com/fastapi/fastapi/issues/12133), wait for an upstream fastapi fix or v0.7.0 on my end

- ✅ **Workaround for [PyPI Releases](https://brokensrc.dev/get/pypi/)**: Downgrade FastAPI `pip install -U fastapi==0.112.4`

- ✅ **Workaround for [Binary Releases](https://brokensrc.dev/get/releases/)**: `./depthflow-cpu-linux-amd64-latest.bin self pip install fastapi==0.112.4`

- ✅ **It's already fixed when [running from source](https://brokensrc.dev/get/source/)**, related commit: https://github.com/BrokenSource/BrokenSource/commit/85e172bae5237328c818db47b7428db3c5c9a50b

<br>

### Explanation

The versions of DepthFlow v0.6.0 asks for [gradio==4.22](https://github.com/BrokenSource/BrokenSource/blob/501c3a4eec85a2359f398af6ccace5d84e24211c/pyproject.toml#L89) ('`==`' as they change it so much), but let it specify a [FastAPI](https://github.com/fastapi/fastapi) version as it wishes; [Gradio](https://files.pythonhosted.org/packages/27/0f/9ebe13938b1357123cffa630468d390d8fbbf344bba27441b97fd491555e/gradio-4.22.0-py3-none-any.whl) simply asks for `'fastapi'` dependency (open as a zip and go to `/gradio-4.22.0.dist-info/METADATA`) and it'll download latest one (0.144.2 at the time)

However, recent versions of it, more specifically later versions of v0.112.4 (release Sep 5) introduced a regression https://github.com/fastapi/fastapi/issues/12133 and apparently broke Path inputs on Gradio apps

- It's updated and fixed on git main, not much I can do for pypi or binary releases but wait on future fastapi fixes

<sub><b>Sidenote:</b> I'll pin to "equals" versions of fastapi and other deps in the future, at least for binary releases, to avoid this</sub>

**Comment by curlysasha** [comment]
same

**Comment by Tremeschin** [comment]
I know it's been two weeks, but I'm very close to a 0.7 release after fixing some technical debt (+lots of improvements). I'll overhaul the webui to something useful and write the remaining `examples` doc page, test and push a 0.7 release !

**Comment by Tremeschin** [comment]
I have just pushed a [**v0.7.0.dev1 wheel**](https://pypi.org/project/depthflow/0.7.0.dev1/) on PyPI, and the GitHub Actions workflow just built executable releases for all platforms here: https://github.com/BrokenSource/BrokenSource/actions/runs/11172285564

initial version of a website page that explains what the parameters do (with examples) also available here https://brokensrc.dev/depthflow/learn/parameters/

I'll write the examples section and reorganize some stuff and go for a full 0.7 release, in the mean time anyone interested please help me test those binaries and python packages, break them 😈!

**Comment by Tremeschin** [comment]
Should be fixed on v0.7.0: 
- https://github.com/BrokenSource/DepthFlow/releases/tag/v0.7.0
- https://pypi.org/project/depthflow/

Feedback appreciated! 🙂
